tests/e2e/workflows/workflow_test.go

```diff
@@ -0,0 +1,166 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package workflows_e2e
+
+import (
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/dapr/dapr/tests/e2e/utils"
+	kube "github.com/dapr/dapr/tests/platforms/kubernetes"
+	"github.com/dapr/dapr/tests/runner"
+	"github.com/stretchr/testify/require"
+)
+
+type testCommandRequest struct {
+	Message string `json:"message,omitempty"`
+}
+
+type appResponse struct {
+	Message   string `json:"message,omitempty"`
+	StartTime int    `json:"start_time,omitempty"`
+	EndTime   int    `json:"end_time,omitempty"`
+}
+
+const numHealthChecks = 60 // Number of times to check for endpoint health per app.
+
+var tr *runner.TestRunner
+
+func TestMain(m *testing.M) {
+	utils.SetupLogs("workflowtestdapr")
+	utils.InitHTTPClient(true)
+
+	// This test shows how to validate the side-car injection
+	// and validate the response by using test app's service endpoint
+
+	// These apps will be deployed for workflowdapr test before starting actual test
+	// and will be cleaned up after all tests are finished automatically
+	testApps := []kube.AppDescription{
+		{
+			AppName:           "workflowsapp",
+			DaprEnabled:       true,
+			ImageName:         "e2e-workflowsapp",
+			Replicas:          1,
+			IngressEnabled:    true,
+			MetricsEnabled:    true,
+			DaprMemoryLimit:   "200Mi",
+			DaprMemoryRequest: "100Mi",
+			AppMemoryLimit:    "200Mi",
+			AppMemoryRequest:  "100Mi",
+			AppPort:           3000,
+		},
+	}
+
+	tr = runner.NewTestRunner("workflowsapp", testApps, nil, nil)
+	os.Exit(tr.Start(m))
+}
+
+var workflowAppTests = []struct {
+	in               string
+	app              string
+	testCommand      string
+	expectedResponse string
+}{
+	{
+		"workflow dapr",
+		"workflowsapp",
+		"start",
+		"Completed",
+	},
+}
+
+func TestWorkflow(t *testing.T) {
+	for _, tt := range workflowAppTests {
+		t.Run(tt.in, func(t *testing.T) {
+			// Get the ingress external url of test app
+			externalURL := tr.Platform.AcquireAppExternalURL(tt.app)
+			require.NotEmpty(t, externalURL, "external URL must not be empty")
+
+			// Check if test app endpoint is available
+			_, err := utils.HTTPGetNTimes(externalURL, numHealthChecks)
+			require.NoError(t, err)
+
+			postString := fmt.Sprintf("%s/StartWorkflow/dapr/placeOrder/e2eInstanceId", externalURL)
+
+			// Sleep so that the workflow engine starts
+			time.Sleep(10 * time.Second)
+
+			// Start the workflow
+			resp, err := utils.HTTPPost(postString, nil)
+			require.NoError(t, err)
+
+			// Assert that the workflow instanceID is returned
+			require.Equal(t, "e2eInstanceId", string(resp))
+
+			getString := fmt.Sprintf("%s/dapr/e2eInstanceId", externalURL)
+			resp, err = utils.HTTPGet(getString)
+
+			// Assert that the status returned is Running
+			require.Equal(t, "Running", string(resp))
+
+			// Pause the workflow
+			postString = fmt.Sprintf("%s/PauseWorkflow/dapr/e2eInstanceId", externalURL)
```

Comment:
Please break this large test into multiple sub-tests. Like, multiple: `t.Run` such as `t.Run("pause and resume workflow")`